### PR TITLE
grc: add support for absolute store paths

### PR DESCRIPTION
**Pull request by azahi**
## Description of changes

Add support for absolute paths from Nix store. This will allow grc to apply configs to `/nix/store/*-package-1.0.0/bin/hello` executables.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/grc/default.nix [hunk 1]
```diff
@@ -21,6 +21,10 @@ buildPythonApplication rec {
       substituteInPlace $f \
         --replace /usr/local/ $out/
     done
+
+    # Support for absolute store paths.
+    substituteInPlace grc.conf \
+      --replace "^([/\w\.]+\/)" "^([/\w\.\-]+\/)"
```
Comment (Artturin) [inline pkgs/tools/misc/grc/default.nix:27]: Needs a better comment/commit message explaining what is being done

**Comment by Artturin** [comment]
`grc $(chase $(whence ps)) aux`
and
`grc ps aux`

Have colors

```
$ whence ps
/etc/profiles/per-user/artturin/bin/ps
grc $(whence ps) aux
```

Doesn't but it's not a regression

Would be nice to tweak and upstream

**Comment by Izorkin** [comment]
Is it possible to add a patch that changes the blue color to a brighter blue?
Example:
```diff
diff --git a/pkgs/tools/misc/grc/default.nix b/pkgs/tools/misc/grc/default.nix
index 19e926c5caf6..9fde688a3108 100644
--- a/pkgs/tools/misc/grc/default.nix
+++ b/pkgs/tools/misc/grc/default.nix
@@ -25,6 +25,11 @@ buildPythonApplication rec {
     # Support for absolute store paths.
     substituteInPlace grc.conf \
       --replace "^([/\w\.]+\/)" "^([/\w\.\-]+\/)"
+
+    # Replace blue color
+    substituteInPlace colourfiles/conf.* \
+      --replace "colours=blue" "colours=bright_blue" \
+      --replace "colours=default,blue" "colours=default,bright_blue"
   '';

   nativeBuildInputs = [ installShellFiles ];
```
Reason: оn a dark background it is very difficult to read text written in blue. You have to strain your eyesight.

**Comment by peterhoeg** [comment]
┌────
│ {
│   environment.systemPackages = [
│     (grc.overrideAttrs (old: {
│       postPatch = old.postPatch + ''
│         for f in colourfiles/conf.* ; do
│           substituteInPlace $f --replace what-ever you-want-it-to-be
│         done
│       '';
│     }))
│   ];
│ }
└────

The better way is of course to patch upstream to look in `/etc/grc' (or similar) for additional configuration files, so you can override things there.

**Comment by Izorkin** [comment]
Maybe then add an option - `replaceBlueColor ? false`.
Then it will be easier to override the configuration and anyone experiencing the same complexity can take advantage of this?

**Comment by azahi** [comment]
@Izorkin This PR is completely unrelated to your request. Please create a separate issue or a PR if you want to see this added. 

It also seems like what you want is purely user-specific and depends entirely on the colourscheme, I'm not sure if we should patch something like this in Nixpkgs. I think the best solution here for you is to override the package locally just for yourself like in Peter's suggestion.

**Pull request merged**